lib/Fhp/Action/GetDepot.php

```diff
@@ -0,0 +1,171 @@
+        ]);
+    }
+
+    public function unserialize($serialized)
+    {
+        list(
+            $parentSerialized,
+            $this->account,
+            ) = unserialize($serialized);
+        parent::unserialize($parentSerialized);
+    }
+
+    /**
+     * @return string The raw MT535 data received from the server.
+     * @noinspection PhpUnused
+     */
+    public function getRawMT535(): string
+    {
+        $this->ensureDone();
+        return $this->rawMT535;
+    }
+
+    /**
+     * @return array The parsed MT535 data.
+     */
+    public function getParsedMT535(): array
+    {
+        $this->ensureDone();
+        return $this->parsedMT535;
+    }
+
+    /**
+     * @return StatementOfAccount
+     */
+    public function getStatement()
+    {
+        $this->ensureDone();
+        return $this->statement;
+    }
+
+    /** {@inheritdoc} */
+    protected function createRequest(BPD $bpd, ?UPD $upd)
+    {
+
+        /** @var HIWPDS $hiwpds */
+        $hiwpds = $bpd->requireLatestSupportedParameters('HIWPDS');
+
+        switch ($hiwpds->getVersion()) {
+            case 5:
+                return HKWPDv5::create(KtvV3::fromAccount($this->account));
+            default:
+                throw new UnsupportedException('Unsupported HKWPD version: ' . $hiwpds->getVersion());
+        }
+		print_r($hiwpds);
+    }
+
+    /** {@inheritdoc} */
+    public function processResponse(Message $response)
+    {
+		
+
+        parent::processResponse($response);
+
+        $isUnavailable = $response->findRueckmeldung(Rueckmeldungscode::NICHT_VERFUEGBAR) !== null;
+        $responseHiwpd = $response->findSegments(HIWPDv5::class);
+		
+        $numResponseSegments = count($responseHiwpd);
+        if (!$isUnavailable && $numResponseSegments < count($this->getRequestSegmentNumbers())) {
+            throw new UnexpectedResponseException("Only got $numResponseSegments HIWPD response segments!");
+        }
+
+        /** @var HIWPD $hiwpd */
+        foreach ($responseHiwpd as $hiwpd) {
+            $this->rawMT535 .= $hiwpd->getDepotaufstellung()->getData();
+        }
+
+        // Note: Pagination boundaries may cut in the middle of the MT535 data, so it is not possible to parse a partial
+        // reponse before having received all pages.
+        if (!$this->hasMorePages()) {
+            $this->parseMt535();
+        }
+    }
+
+    private function parseMt535()
+    {
+        if (strpos(strtolower($this->bankName), 'sparda') !== false) {
+            $parser = new SpardaMT535();
+        } elseif (strpos(strtolower($this->bankName), 'postbank') !== false) {
+            $parser = new PostbankMT535();
+        } else {
+            $parser = new MT535();
+        }
+
+        try {
+            // Note: Some banks encode their MT 535 data as SWIFT/ISO-8859 like it should be according to the
+            // specification (e.g. DKB), others just send UTF-8 (e.g. Consorsbank), so we try to detect it here.
```

Comment:
Remove the `e.g.`s here, as these particular banks might not even support MT535. Detecting the encoding is still a good idea.